nitro-protocol/contracts/NitroAdjudicator.sol

```diff
@@ -13,34 +13,28 @@ contract NitroAdjudicator is ForceMove, MultiAssetHolder {
     /**
      * @notice Finalizes a channel by providing a finalization proof, and liquidates all assets for the channel.
      * @dev Finalizes a channel by providing a finalization proof, and liquidates all assets for the channel.
-     * @param largestTurnNum The largest turn number of the submitted states; will overwrite the stored value of `turnNumRecord`.
      * @param fixedPart Data describing properties of the state channel that do not change with state updates.
-     * @param appData Application specific data.
-     * @param outcomeBytes abi.encode of an array of Outcome.OutcomeItem structs.
+     * @param latestVariablePart Latest variable part in finalization proof. Must have the largest turnNum and the same appData and outcome as all other variable parts in finalization proof.
      * @param numStates The number of states in the finalization proof.
      * @param whoSignedWhat An array denoting which participant has signed which state: `participant[i]` signed the state with index `whoSignedWhat[i]`.
      * @param sigs Array of signatures, one for each participant, in participant order (e.g. [sig of participant[0], sig of participant[1], ...]).
      */
     function concludeAndTransferAllAssets(
-        uint48 largestTurnNum,
         FixedPart memory fixedPart,
-        bytes memory appData,
-        bytes memory outcomeBytes,
+        IForceMoveApp.VariablePart memory latestVariablePart,
         uint8 numStates,
         uint8[] memory whoSignedWhat,
         Signature[] memory sigs
     ) public {
         bytes32 channelId = _conclude(
-            largestTurnNum,
             fixedPart,
-            appData,
-            outcomeBytes,
+            latestVariablePart,
             numStates,
             whoSignedWhat,
             sigs
         );
 
-        transferAllAssets(channelId, outcomeBytes, bytes32(0));
+        transferAllAssets(channelId, Outcome.encodeExit(latestVariablePart.outcome), bytes32(0));
```

Comment:
There are different options here, but to me it seems slightly nicer to:

* pass the `latestVariablePart.outcome`
* have `transferAllAssets` encode and hash it

(rather than encoding it here, passing the bytes, and having `transferAllAssets` hash it and decode it).

Reply:
This way looks better to me too.